### DISCUSSION TITLE
Passing argument to predefine component style

### DISCUSSION THREAD

**Discussion by kayac-chang** [tailwindlabs/tailwindcss#8662] [Help] [answered]
Hello!!!
Is it possible to define a dynamic argument based on predefine component ?
I have some use case below.

## separate

```css
@layer components {
  .separate-[/* argument */] {
    @apply [&>*:not(:last-child)]:after:content-[/* argument */];
    @apply [&>*:not(:last-child)]:after:inline-block;
    @apply [&>*:not(:last-child)]:after:mx-2;
  }
}
```
which can generate separate token between each children element
### Example
```
<ul class="separate-[>]">
  <li>🍎</li>
  <li>🥟</li>
  <li>🎉</li>
</ul>

=> 🍎 > 🥟 > 🎉
```

**Comment by timohoelzer** [comment, marked as answer]
Hey kayac-chang,

not really I think. 

Multiple classes is one way, that even brings more flexibilty styling different seperators. 
But if you want to keep things low profile CSS you could use CSS variables as a flexible approach:

**TAILWIND**
```
@layer components {
    .separate, .separate li{
        @apply flex gap-4;
    }
    .separate li:not(:last-child):after{
      content: var(--seperator,'–');
    }
 }
```

**HTML**
```
<ul class="separate" style="--seperator:'>'">
    <li>🍎</li>
    <li>🥟</li>
    <li>🎉</li>
  </ul>
```

Here's a codepen to see it in action / play with
https://codepen.io/tatundkraft/pen/eYVXoMr/c089e1f576f33b4ea9ca7438ed25a7d5?editors=1100

**Reply by kayac-chang** [reply]
awesome!!! thank you for the solution.
